keylime/src/list_parser.rs

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Keylime Authors
+
+use pest::{iterators::Pair, Parser};
+use pest_derive::Parser;
+use thiserror::Error;
+
+#[derive(Parser)]
+#[grammar = "list.pest"]
+pub struct ListParser;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Parsing error: {0}")]
+    ParseError(Box<pest::error::Error<Rule>>),
+}
+
+fn get_inner_str(pair: Pair<Rule>) -> Vec<&str> {
+    let mut l = Vec::new();
+    for item in pair.into_inner() {
+        match item.as_rule() {
+            Rule::list
+            | Rule::bracketed
+            | Rule::unbracketed
+            | Rule::name
+            | Rule::bracketed_name
+            | Rule::single
+            | Rule::double => {
+                l.extend(get_inner_str(item));
+            }
+            Rule::single_content
+            | Rule::double_content
+            | Rule::unquoted
+            | Rule::not_closer
+            | Rule::bracketed_unquoted => {
+                let i = item.as_str();
+                if !i.is_empty() {
+                    l.push(i);
+                }
+            }
+            Rule::EOI => {
+                break;
+            }
+            _ => {
+                unreachable!()
+            }
+        }
+    }
+    l
+}
+
+/// Parses a list from a string slice and return a Vec<&str>
+///
+/// The list in the input string:
+///
+/// * can contain single-quoted, double-quoted, or unquoted strings
+/// * can be inside square brackets ("[]") or not
+/// * can have string separated by commas, white spaces, or newlines
+///
+/// # Arguments
+///
+/// * `list` the string to be parsed
+///
+/// # Returns
+///
+/// The obtained list as a Vec<&str>
+///
+/// # Examples
+///
+/// Valid input lists:
+///
+/// * `a b c`
+/// * `'a' "b" c`
+/// * `'a', 'b', 'c'`
+/// * `[a b c]`
+/// * `['a', "b", c]`
+///
```

Comment:
This interface looks a bit too generic to be exposed from the `keylime` lib crate. Not necessarily in this PR, but we might eventually want to move the entire configuration handling in the lib crate with access methods to retrieve list of certs, actions, etc.

Reply:
I noticed a bug in the way I parsed the strings: I cannot remove the quotes from quoted strings. I made the changes to keep the quotes.